projects/Button_ADC/src/main.c

```diff
@@ -0,0 +1,53 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress button_address = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioAddress adc_address = { .port = GPIO_PORT_A, .pin = 6 };
+
+static GpioSettings adc_settings = {
+  .direction = GPIO_DIR_IN,
+  .resistor = GPIO_RES_NONE,
+  .alt_function = GPIO_ALTFN_ANALOG,
+};
+
+static InterruptSettings interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+static AdcChannel adc_pinA6_channel = ADC_CHANNEL_6;
+
+static uint16_t channel_data = 0;
+
+static void prv_button_interrupt_handler(const GpioAddress *address, void *context) {
+  adc_read_raw(adc_pinA6_channel, &channel_data);
```

Comment:
The FW 103 homework directions do say:
> Create a project that reads an ADC **converted** reading...

This is a raw reading, look in adc.h for more details.